core/math/a_star.cpp

```diff
@@ -201,9 +211,8 @@ bool AStar::are_points_connected(int p_id, int p_with_id) const {
 
 void AStar::clear() {
 
-	for (const Map<int, Point *>::Element *E = points.front(); E; E = E->next()) {
-
-		memdelete(E->get());
+	for (OAHashMap<int, Point *>::Iterator it = points.iter(); it.valid; it = points.next_iter(it)) {
+		memdelete(*(it.value));
```

Comment:
Wouldn't `points.clear()` do this now?

Reply:
It would call the destructor but it wouldn't necessarily call memdelete on any pointer types that are in there, in this case the memnew'd Points, so this is still necessary.